### PR TITLE
Feat/zustand

### PR DESCRIPTION
**Pull request by isVivek99**
### What is the change?
#126 

added zustand as a state management library

### Is it bug?
No

### \*Dev Tested?

- [x] Yes
- [ ] No

#### Platforms

- [x] Web

#### Browsers

- [x] Chrome

### CODE REVIEW

File: app/routes/index.tsx [hunk 1]
```diff
@@ -2,10 +2,15 @@ import { LinksFunction } from '@remix-run/node';
...
 import Calendar from '~/components/Calendar';
 import Navbar from '~/components/common/navbar';
+import { useBoundStore } from '~/store/useStore';
+import { EventState, UserState } from '~/utils/interfaces';
 
 export const links: LinksFunction = () => [{ rel: 'stylesheet', href: styles }];
 
 export default function Index() {
+  const events = useBoundStore((state: EventState & UserState) => state.events);
+  console.log(events);
```
Comment (sumitd94) [inline app/routes/index.tsx:12]: Can you please remove this?

File: app/store/eventSlice.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable comma-dangle */
+/* eslint-disable implicit-arrow-linebreak */
+/* eslint-disable no-unused-vars */
+/* eslint-disable import/prefer-default-export */
```
Comment (sumitd94) [inline app/store/eventSlice.ts:4]: Why?

File: app/utils/interfaces.ts [hunk 3]
```diff
@@ -27,3 +27,23 @@ export interface UpdateEvent {
...
+  removeEvent: (event: CalEvent) => void;
+}
+
+export interface User {
+  firstname: string;
+  lastname: string;
+  bio: string;
+  username: string;
+  timezone: string;
+  onboarding: boolean;
```
Comment (yesyash) [inline app/utils/interfaces.ts:43]: Nitpick:
As this is a boolean can we rename it to  `onboardingComplete` or `isOnboarded`?

File: app/types/GlobalState.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export type updateUserKey = 'firstname' | 'lastname' | 'bio' | 'username' | 'timezone';
```
Comment (yesyash) [inline app/types/GlobalState.ts:1]: Nitpick: 
Types are generally defined with the first character capitalized.

File: package.json [hunk 5]
```diff
@@ -37,10 +37,11 @@
     "moment": "^2.29.4",
     "react": "^18.2.0",
     "react-big-calendar": "^1.5.2",
-    "react-datepicker": "^4.10.0",
+    "react-datepicker": "^4.11.0",
```
Comment (yesyash) [inline package.json:40]: Was this required?

**Review by yesyash — CHANGES_REQUESTED** [review]

File: app/routes/index.tsx (hunk 1 above)
Comment (isVivek99) [reply app/routes/index.tsx:12]: yes

File: app/types/GlobalState.ts (hunk 4 above)
Comment (isVivek99) [reply app/types/GlobalState.ts:1]: thanks

File: package.json (hunk 5 above)
Comment (isVivek99) [reply package.json:40]: i think yarn did this, should I revert back?

Comment (yesyash) [reply package.json:40]: Looks like this is causing the test to fail, can you check please.

**Comment by harshith-venkatesh** [comment]
As discussed yesterday, please look into reverting any updates other than zustand installation and test the changes

File: package.json (hunk 5 above)
Comment (isVivek99) [reply package.json:40]: yess

File: app/store/eventSlice.ts (hunk 2 above)
Comment (harshith-venkatesh) [reply app/store/eventSlice.ts:4]: Taking theatest pull will resolve the issue

**Review by yesyash — APPROVED** [review]

File: app/routes/index.tsx [hunk 6]
```diff
@@ -2,10 +2,14 @@ import { LinksFunction } from '@remix-run/node';
...
 import Calendar from '~/components/Calendar';
 import Navbar from '~/components/common/navbar';
+import { useStore } from '~/store/useStore';
+import { EventState, UserState } from '~/utils/interfaces';
 
 export const links: LinksFunction = () => [{ rel: 'stylesheet', href: styles }];
 
 export default function Index() {
+  const events = useStore((state: EventState & UserState) => state.events);
+
```
Comment (iRohitGaur) [inline app/routes/index.tsx:12]: we don’t need this in `index.tsx`. We can directly import in the component where we need it.

File: app/store/eventSlice.ts (hunk 2 above)
Comment (iRohitGaur) [inline app/store/eventSlice.ts:4]: take latest pull, these should be gone now

**Review by iRohitGaur — APPROVED** [review]

File: app/routes/index.tsx [hunk 7]
```diff
@@ -2,10 +2,14 @@ import { LinksFunction } from '@remix-run/node';
 import styles from 'react-datepicker/dist/react-datepicker.css';
 import Calendar from '~/components/Calendar';
 import Navbar from '~/components/common/navbar';
+import { useStore } from '~/store/useStore';
+import { EventState, UserState } from '~/utils/interfaces';
```
Comment (iRohitGaur) [inline app/routes/index.tsx:6]: We don't need the imports as well if we are not using the store here

File: app/routes/index.tsx [hunk 8]
```diff
@@ -2,10 +2,14 @@ import { LinksFunction } from '@remix-run/node';
...
 
 export const links: LinksFunction = () => [{ rel: 'stylesheet', href: styles }];
 
 export default function Index() {
+ 
+
```
[the remark this answers is not in the document]
Comment (isVivek99) [reply app/routes/index.tsx:12]: this is supposed tp be done by the formatter, cannot consider it a nitpick

[the remark this answers is not in the document]
Comment (iRohitGaur) [reply app/routes/index.tsx]: ah, I agree. Not sure why that is happening. For now lets remove it ourselves and report it to Sumit

**Review by iRohitGaur — APPROVED** [review]

File: app/routes/index.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (iRohitGaur) [reply app/routes/index.tsx]: CC: @sumitd94

**Review by harshith-venkatesh — APPROVED** [review]
Thanks @isVivek99 for quickly integrating zustand with this project.

**Pull request merged**